analyze-trace.ts

```diff
@@ -162,18 +193,41 @@ async function main(): Promise<void> {
         }
     }
 
-    await printHotStacks(root);
+    await makeHotStacks(root);
+}
+
+type TreeNode = {
+    type: string
+    time?: string
+    message: string
+    terseMessage: string
+    start?: {
+        file: string
+        offset?: number
+    },
+    end?: {
+        file: string
+        offset?: number
+    }
+    children: TreeNode[]
 }
 
-async function printHotStacks(root: EventSpan): Promise<void> {
+async function makeHotStacks(root: EventSpan): Promise<void> {
     if (typesPath) {
         await addTypeTrees(root);
     }
+
     const positionMap = await getNormalizedPositions(root);
+
     const tree = await makePrintableTree(root, /*currentFile*/ undefined, positionMap);
-    if (Object.entries(tree).length) {
+    
+    if (tree && Object.entries(tree).length) {    
+        if (opts.json) {
```

Comment:
Intuitively, I would have expected json output to _replace_ the normal output, not supplement it.